pipeline/upstream_listener.groovy

```diff
@@ -0,0 +1,82 @@
+/*
+    Script that updates upstream recipe file 
+*/
+// Global variables section
+def nodeName = "centos-7"
+def sharedLib
+
+// Pipeline script entry point
+node(nodeName) {
+    try {
+        timeout(unit: "MINUTES", time: 30) {
+            stage('Preparing') {
+                if (env.WORKSPACE) { sh script: "sudo rm -rf * .venv" }
+                checkout(
+                    scm: [
+                        $class: 'GitSCM',
+                        branches: [[name: 'origin/master']],
+                        extensions: [
+                            [
+                                $class: 'CleanBeforeCheckout',
+                                deleteUntrackedNestedRepositories: true
+                            ],
+                            [
+                                $class: 'WipeWorkspace'
+                            ],
+                            [
+                                $class: 'CloneOption',
+                                depth: 1,
+                                noTags: true,
+                                shallow: true,
+                                timeout: 10,
+                                reference: ''
+                            ]
+                        ],
+                        userRemoteConfigs: [[
+                            url: 'https://github.com/red-hat-storage/cephci.git'
+                        ]]
+                    ],
+                    changelog: false,
+                    poll: false
+                )
+                sharedLib = load("${env.WORKSPACE}/pipeline/vars/lib.groovy")
+                sharedLib.prepareNode()
+            }
+        }
+
+        stage('Updating') {
```

Comment:
can we introduce few retries with some delay, if script could not able to update release.yaml?

Reply:
Done. Added 3 retries. Sunil angadi's already taking care of sleep between retry